### ISSUE TITLE
Creating old movement logs disrupts current location logic

### ISSUE THREAD

**Issue by botlfarm** [farmOS/farmOS#886]
Open an asset that has a location. Create a movement log in the past. The current location is updated to this old log location not the actual location. 
Screen recording for clarity. https://drive.google.com/file/d/11pesiLqE-I-nr9-K-mEzbNcthuNNu1n5/view?usp=sharing

version 3.2.1

**Comment by botlfarm** [comment]
I should add that this is an asset in a group and typically moves with group movement logs.

**Comment by mstenta** [comment]
Confirming this. @botlfarm and I observed it on a live instance together, in addition to the instance in the video recording linked above.

However, I tried replicating in my local instance and was unable to. I'll need to do some more digging to figure out what's going on... :thinking:

**Comment by mstenta** [comment]
Disabling JavaScript does not solve the problem. So the elements seem to be hidden in the server-generated HTML/CSS.

**Comment by mstenta** [comment]
Ah! I made Claro the default theme (instead of Gin) and now it's working. So this may be a Gin issue.

**Comment by mstenta** [comment]
I just did a clean test of this locally and was able to replicate it. I did it all via Drush commands to document the exact steps...

First, install farmOS with all the default modules:

`drush site-install -y --db-url=pgsql://farm:farm@db/farm --account-pass=admin`

Enable the Group asset module:

`drush en -y farm_group`

Open a PHP CLI prompt with `drush php:cli` and paste the following code:

```
$asset_storage = \Drupal::entityTypeManager()->getStorage('asset');
$log_storage = \Drupal::entityTypeManager()->getStorage('log');
$term_storage = \Drupal::entityTypeManager()->getStorage('taxonomy_term');

// Create two fields as land assets.
$land1 = $asset_storage->create([
  'name' => 'Field A',
  'type' => 'land',
  'land_type' => 'field',
]);
$land1->save();
$land2 = $asset_storage->create([
  'name' => 'Field B',
  'type' => 'land',
  'land_type' => 'field',
]);
$land2->save();

// Create a Sheep animal asset.
$animal_type = $term_storage->create([
  'name' => 'Sheep',
  'vid' => 'animal_type',
]);
$animal_type->save();
$animal = $asset_storage->create([
  'name' => 'Sheep',
  'type' => 'animal',
  'animal_type' => [$animal_type],
]);
$animal->save();

// Create a Sheep Herd group asset.
$group = $asset_storage->create([
  'name' => 'Sheep Herd',
  'type' => 'group',
]);
$group->save();

// Create an observation log that makes Sheep a member of the Sheep Herd.
$log = $log_storage->create([
  'name' => 'Group Sheep into Sheep Herd',
  'type' => 'observation',
  'asset' => [$animal],
  'group' => [$group],
  'is_group_assignment' => TRUE,
  'status' => 'done',
]);
$log->save();

// Create an activity log that moves Sheep Herd to Field A.
$log = $log_storage->create([
  'name' => 'Move Sheep Herd to Field A',
  'type' => 'activity',
  'asset' => [$group],
  'location' => [$land1],
  'is_movement' => TRUE,
  'status' => 'done',
]);
$log->save();

// Create an activity log that moves Sheep to Field B last year.
$log = $log_storage->create([
  'timestamp' => strtotime('-1 year'),
  'name' => 'Move Sheep to Field B',
  'type' => 'activity',
  'asset' => [$animal],
  'location' => [$land2],
  'is_movement' => TRUE,
  'status' => 'done',
]);
$log->save();
```

Now you can observe that the "Sheep" asset is in "Field B", even though "Sheep Herd" has a more recent movement log to "Field A". You can either see this through the UI or with the following PHP in the same `drush php:cli` prompt:

`\Drupal::service('asset.location')->getLocation($animal)[0]->label();`

Now to figure out why...

**Comment by mstenta** [comment]
I figured out the cause of this. It is this line in the Group module's asset location service logic:

https://github.com/farmOS/farmOS/blob/09e94797270f7f9ae1d50be9b920736580a721c6/modules/asset/group/src/GroupAssetLocation.php#L98

I think the intention of that line was to mimic the underlying asset location logic for cases where two movement logs have the same timestamp. In that case the log with the higher ID is used. But this line is obviously wrong with that in mind, and is resulting in logs with a higher ID (even those in the distant past) being picked as the "latest movement" for the asset.

So it should be a pretty easy fix. :-)

**Comment by mstenta** [comment]
I think the logic we WANT is this:

`if (($group_movement_timestamp > $latest_timestamp) || ($group_movement_timestamp == $latest_timestamp && $group_movement_id > $latest_id)) {`

(aka: "If the group membership timestamp is greater than the latest log timestamp of the asset OR if it's the same but the log ID is greater")

**Comment by mstenta** [comment]
Opened a pull request with a fix and a regression test: #888
